packages/core/src/shared/clients/ec2Client.ts

```diff
@@ -13,9 +13,13 @@ import { Timeout } from '../utilities/timeoutUtils'
 import { showMessageWithCancel } from '../utilities/messages'
 import { ToolkitError, isAwsError } from '../errors'
 
-export interface Ec2Instance extends EC2.Instance {
-    name?: string
-    status?: EC2.InstanceStateName
+/**
+ * A wrapper around EC2.Instance where we can safely assume InstanceId field exists.
+ */
+export interface SafeEc2Instance extends EC2.Instance {
```

Comment:
This is a common pattern and we did it for CodeCatalyst. We can just re-use the same names, don't need to add things like `Safe`. If there is ever a "conflict", modules can just namespace their imports:

    import * as ec2Client
    const foo: ec2Client.Ec2Instance